File: java/P003.java
/*
 * Determine largest prime factor of 600851475143
 */
public final class P003 implements Solution {

	public static void main(String[] args) {
		System.out.println(new P003().run());
	}

	public String run() {
		long n = 600851475143L;
		int largest_factor = 1;
		while (n > 1) {
			boolean found = false ;
			int max_i = (int) Math.sqrt(n);
			for (int i = 2; i <= max_i; i++) {
				if (n % i == 0) {
					largest_factor = Math.max(i, largest_factor);
					n = n / i ;
					found= true ;
					break;
				}
			}
			if (! found) {
				largest_factor = Math.max((int)n,largest_factor);
				n = 1;
			}
		}
		return Integer.toString(largest_factor);
	}

}

File: python/P003.py
def P003()->int:
    """Determine largest prime factor of 600851475143"""
    n = 600851475143 # input
    factor = 1       # container for result

    # Iteratively peel off prime factors until we've factorized n
    while n > 1:
        found = False
        for i in range(2, int(n ** 0.5)): # if we don't find something before sqrt(n), then n is prime
            if n % i == 0: # n is composite
                factor = max(i, factor)
                n = n // i # try again with reduced input
                found = True
                break
        if not found: # n is prime - we're done factorizing!
            factor = max(factor, n)
            break
    return factor

if __name__=="__main__":
    print(P003())
